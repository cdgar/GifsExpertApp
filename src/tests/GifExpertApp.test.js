import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('GifExpertApp', () => {

    test('should show GifExpertApp', () => {
        const wrapper = shallow(<GifExpertApp />)
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show GifExpertApp with list', () => {
        const categories = ['dragon', 'red'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
    
})