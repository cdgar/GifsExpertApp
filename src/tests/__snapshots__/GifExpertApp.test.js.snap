// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should show GifExpertApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp!
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;

exports[`GifExpertApp should show GifExpertApp with list 1`] = `
<Fragment>
  <h2>
    GifExpertApp!
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="dragon"
      key="dragon"
    />
    <GifGrid
      category="red"
      key="red"
    />
  </ol>
</Fragment>
`;
